tensorboard/plugins/base_plugin.py

```diff
@@ -122,10 +126,22 @@ def __init__(
           logic should handle cases in which a plugin is absent from this
           mapping, lest a KeyError is raised.
       window_title: A string specifying the the window title.
+
+    Raises:
+      ValueError: If both logdir and db_uri are provided (or if neither are
+          provided).
     """
+    if db_uri and logdir:
```

Comment:
Might not be worth adding hardening business logic to magic container. If statements like these might be more appropriate for all the other business logic that interacts with the magic container.

Reply:
Done. SG - the exceptions don't guard against later changing the properties anyway.